ui/lib/CreateStream.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, ReactNode } from "react";
+import Slider from "rc-slider";
+import { Address, WalletClient } from "viem";
+import {
+  useAccount,
+  useBalance,
+  usePublicClient,
+  useWalletClient,
+  usePrepareContractWrite,
+  useContractWrite,
+} from "wagmi";
+
+import StreamManager, { Stream } from "@apeworx/apepay";
+
+const SECS_PER_DAY = 24 * 60 * 60;
+
+export interface CreateStreamProps {
+  streamManagerAddress: Address;
+  // TODO: Support dynamically fetching list of accepted tokens in sdk::StreamManager
+  tokenAddress: Address;
+  amountPerSecond: number;
+  reasonCode: string;
+  cart?: ReactNode;
+  registerStream: (stream: Stream) => void;
+}
+
+const CreateStream = (props: CreateStreamProps) => {
+  const { address } = useAccount();
+  const { data: tokenData } = useBalance({
+    address,
+    token: props.tokenAddress,
+  });
+  // TODO: handle `isError`, `isLoading`
+  const maxTime = Number(
+    (tokenData?.value || BigInt(0)) / BigInt(props.amountPerSecond),
+  );
+  // TODO: Increase stability of deployments beyond a week
+  const maxTimeDays: number = Math.min(Math.floor(maxTime / SECS_PER_DAY), 7); // Up to a week
+  const marks = Object.fromEntries(
+    Array.from(Array(maxTimeDays).keys()).map((v: number) => [
+      v + 1,
+      `${v + 1}`,
+    ]),
+  );
+  const [selectedTime, setSelectedTime] = useState(SECS_PER_DAY); // Defaults 1 day
+
+  const sm = new StreamManager(
+    props.streamManagerAddress,
+    // TODO: handle `isError`, `isLoading`
+    usePublicClient(),
+    useWalletClient()?.data as WalletClient,
+  );
+
+  const { config: approvalConfig } = usePrepareContractWrite({
+    address: props.tokenAddress,
+    value: BigInt(0),
+    abi: [
+      {
+        name: "approve",
+        type: "function",
+        stateMutability: "nonpayable",
+        inputs: [
+          { name: "spender", type: "address" },
+          { name: "amount", type: "uint256" },
+        ],
+        outputs: [{ name: "success", type: "bool" }],
+      },
+    ],
+    functionName: "approve",
+    args: [props.streamManagerAddress, selectedTime * props.amountPerSecond],
+  });
+  const { write: approveStream } = useContractWrite(approvalConfig);
+
+  const createStream = async () => {
+    const stream = await sm.create(
+      props.tokenAddress,
+      props.amountPerSecond,
+      props.reasonCode,
+    );
+
+    // NOTE: This function should move away from this component
+    props.registerStream(stream);
+  };
+
+  return (
+    <div id="CreateStream">
+      {/* Display: What is the product they are paying for? With what token? */}
+      {props.cart && props.cart}
```

Comment:
This line is causing issues for me, unless I comment it out.

Reply:
It was a way for me to provide a space for displaying a "cart" (which could be compressed down to form the reason string) but yeah, I'm leaning more towards just providing wagmi-style react hooks with this library and moving the UI elements to the demo app, so people can choose their own components when installing